File: src/main/java/com/jss/jssbackend/modules/tiers/service/TiersCategoryService.java
package com.jss.jssbackend.modules.tiers.service;

import java.util.List;

import com.jss.jssbackend.modules.tiers.model.TiersCategory;

public interface TiersCategoryService {
    public List<TiersCategory> getTiersCategories();

    public TiersCategory getTiersCategory(Integer id);
}

File: client-osiris/src/app/modules/tiers/services/tiers.category.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { TiersCategory } from '../../profile/model/TiersCategory';

@Injectable({
  providedIn: 'root'
})
export class TiersCategoryService extends AppRestService<TiersCategory> {

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getTiersCategories() {
    return this.getList(new HttpParams(), "tiers-categories");
  }

}
